src/test/java/com/iota/iri/service/TipsManagerTest.java

```diff
@@ -65,111 +75,269 @@ public static void tearDown() throws Exception {
+        transaction4 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction2.getHash(),
+                transaction3.getHash()), getRandomTransactionHash());
         transaction.store(tangle);
         transaction1.store(tangle);
         transaction2.store(tangle);
-        Map<Hash, Set<Hash>> ratings = new HashMap<>();
-        tipsManager.updateHashRatings(transaction.getHash(), ratings, new HashSet<>());
-        Assert.assertEquals(ratings.get(transaction.getHash()).size(), 3);
-        Assert.assertEquals(ratings.get(transaction1.getHash()).size(), 2);
-        Assert.assertEquals(ratings.get(transaction2.getHash()).size(), 1);
+        transaction3.store(tangle);
+        transaction4.store(tangle);
+        Map<Buffer, Integer> txToCw = tipsManager.calculateCumulativeWeight(new HashSet<>(),
+                transaction.getHash(), false, new HashSet<>());
+
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 4),
+                1, txToCw.get(transaction4.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 3),
+                2, txToCw.get(transaction3.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 2),
+                3, txToCw.get(transaction2.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 1),
+                4, txToCw.get(transaction1.getHash().getSubHash()).intValue());
+        Assert.assertEquals(String.format(TX_CUMULATIVE_WEIGHT_IS_NOT_AS_EXPECTED_FORMAT, 0),
+                5, txToCw.get(transaction.getHash().getSubHash()).intValue());
     }
 
     @Test
-    public void updateRatingsTestWorks() throws Exception {
+    public void testCalculateCumulativeWeightDiamond() throws Exception {
+        TransactionViewModel transaction, transaction1, transaction2, transaction3;
+        transaction = new TransactionViewModel(getRandomTransactionTrits(), getRandomTransactionHash());
+        transaction1 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction.getHash(),
+                transaction.getHash()), getRandomTransactionHash());
+        transaction2 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction.getHash(),
+                transaction.getHash()), getRandomTransactionHash());
+        transaction3 = new TransactionViewModel(getRandomTransactionWithTrunkAndBranch(transaction1.getHash(),
+                transaction2.getHash()), getRandomTransactionHash());
+        transaction.store(tangle);
+        transaction1.store(tangle);
+        transaction2.store(tangle);
+        transaction3.store(tangle);
+        log.debug("printing transaction in diamond shape \n                      {} \n{}  {}\n                      {}",
```

Comment:
Do we need this log in a UT?

Reply:
I think it is helpful to log. If others will agree with you I am fine with removing